### PR TITLE
 Backport GHA Migration to release-11.1 

### PR DESCRIPTION
**Pull request by gokhangulbiz**
DESCRIPTION: PR description that will go into the change log, up to 78 characters

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -0,0 +1,465 @@
...
+    runs-on: ubuntu-20.04
+    container:
+      image: ${{ needs.params.outputs.fail_test_image_name }}:${{ needs.params.outputs.pg13_version }}${{ needs.params.outputs.image_suffix }}
+      options: --user root
+    env:
+      runs: 8
+    needs:
+    - params
+    - build
+    - test-flakyness-pre
```
Comment (aykut-bozkurt) [inline .github/workflows/build_and_test.yml:445]: Shouldn't be already required by dependency transition?

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -0,0 +1,465 @@
...
+        else
+          echo "Detected tests " $tests
+        fi
+        echo tests="$tests" >> "$GITHUB_OUTPUT"
+  test-flakyness:
+    if: ${{ needs.test-flakyness-pre.outputs.tests != ''}}
+    name: Test flakyness
+    runs-on: ubuntu-20.04
+    container:
+      image: ${{ needs.params.outputs.fail_test_image_name }}:${{ needs.params.outputs.pg13_version }}${{ needs.params.outputs.image_suffix }}
```
Comment (aykut-bozkurt) [inline .github/workflows/build_and_test.yml:438]: In circleci, we ran on pg15. Isn't it?

Comment (gokhangulbiz) [reply .github/workflows/build_and_test.yml:438]: Yes, it is.

File: .github/workflows/build_and_test.yml (hunk 1 above)
Comment (gokhangulbiz) [reply .github/workflows/build_and_test.yml:445]: I've had problems in the past when I didn't clearly list dependencies. To play it safe, it's a good idea to be explicit about them.

**Review by aykut-bozkurt — APPROVED** [review]
LGTM

**Pull request merged**